setup.py

```diff
@@ -48,14 +48,12 @@ def is_rtd() -> bool:
 install_requires = [
     "overrides",
     "PyYAML",
-    "pydantic>=1.9.0,<2.0.0",
-    "pydantic-yaml[pyyaml]>=0.11.0,<1.0.0",
+    "pydantic>=2.0.0,<3.0.0",
+    "pydantic-yaml[ruamel]>=1.1.0,<=1.2.0",
```

Comment:
This pydantic2 bump is going to happen to all other craft-* repos as well, so there is going to be a re-release for all.
Can you point out in what area will/can ruamel's different behavior affect the *craft projects?

Reply:
what's the need to go from `pyyaml` to `ruamel`?

Reply:
The main thing I can think of is that `ruamel.yaml` uses YAML 1.2 rather than 1.1, and they're not strictly compatible. For example, pyyaml will accept strings like "yes" as booleans, whereas ruamel.yaml keeps them as strings

```
Python 3.11.6 (main, Oct  8 2023, 05:06:43) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import yaml as pyyaml
>>> pyyaml.safe_load("my-bool: yes")
{'my-bool': True}
>>> from ruamel.yaml import YAML
>>> ruamel_yaml = YAML(typ="safe")
>>> ruamel_yaml.load("my-bool: yes")
{'my-bool': 'yes'}
>>> ruamel_yaml.load("my-bool: true")
{'my-bool': True}
```

Personally I prefer the `ruamel` behaviour, but it would be good to put guardrails on the error messages at minimum.

Reply:
I'm not sure if this particular case needs extra guardrails, since it seems that a `yes` is expected to be string `"yes"` anyway in our codebase (see [this example](https://github.com/canonical/craft-parts/blob/0f5ec0ee704ef462e007437bce6c4749b318fab6/tests/unit/plugins/test_plugins.py#L164)).